core/src/main/java/com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.java

```diff
@@ -0,0 +1,95 @@
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+
+import com.linecorp.armeria.common.AggregatedHttpMessage;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.MediaType;
+
+/**
+ * A default implementation of a {@link RequestConverterFunction} which converts a JSON body of
+ * the {@link AggregatedHttpMessage} to an object by {@link ObjectMapper}.
+ */
+public class JacksonRequestConverterFunction implements RequestConverterFunction {
+
+    private static final Logger logger = LoggerFactory.getLogger(JacksonRequestConverterFunction.class);
+
+    private final ObjectMapper mapper;
+    private final ConcurrentMap<Class<?>, ObjectReader> readers = new ConcurrentHashMap<>();
+
+    /**
+     * Creates an instance with the default {@link ObjectMapper}.
+     */
+    public JacksonRequestConverterFunction() {
+        this(new ObjectMapper());
+    }
+
+    /**
+     * Creates an instance with the specified {@link ObjectMapper}.
+     */
+    public JacksonRequestConverterFunction(ObjectMapper mapper) {
+        this.mapper = requireNonNull(mapper, "mapper");
+    }
+
+    /**
+     * Returns whether the specified {@link AggregatedHttpMessage} is able to be consumed.
+     */
+    @Override
+    public boolean accept(AggregatedHttpMessage request, Class<?> expectedResultType) {
+        final String contentType = request.headers().get(HttpHeaderNames.CONTENT_TYPE);
+        // TODO(hyangtack) Do benchmark tests to decide whether we add a cache to MediaType#parse.
+        if (contentType != null &&
+            MediaType.parse(contentType).is(MediaType.JSON)) {
+            try {
+                return readers.computeIfAbsent(expectedResultType, mapper::readerFor) != null;
+            } catch (Throwable cause) {
+                logger.warn(JacksonRequestConverterFunction.class.getName() +
+                            " cannot read a JSON of '" + request.content().toStringUtf8() +
+                            "' as a type '" + expectedResultType.getName() + '\'', cause);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Converts the specified {@link AggregatedHttpMessage} to an object of {@code expectedResultType} type.
+     */
+    @Override
+    public Object convert(AggregatedHttpMessage request, Class<?> expectedResultType) throws Exception {
+        final ObjectReader reader = readers.get(expectedResultType);
+        assert reader != null;
+        final String contentType = request.headers().get(HttpHeaderNames.CONTENT_TYPE);
+        assert contentType != null;
+
+        final Charset charset = MediaType.parse(contentType).charset()
+                                         .orElse(StandardCharsets.ISO_8859_1);
```

Comment:
should be `UTF-8` ?

Reply:
I found the following sentence from [here](https://www.w3.org/International/articles/http-charset/index.en). So I used ISO-8859-1 as the default charset. 
> HTTP 1.1 says that the default charset is ISO-8859-1.

Reply:
I think you are right. UTF-8 is a default charset for JSON. Thanks!

ISO-8859-1 is a default charset for "text" type.
https://tools.ietf.org/html/rfc2616#section-3.7.1
> The "charset" parameter is used with some media types to define the
   character set (section 3.4) of the data. When no explicit charset
   parameter is provided by the sender, media subtypes of the "text"
   type are defined to have a default charset value of "ISO-8859-1" when
   received via HTTP. Data in character sets other than "ISO-8859-1" or
   its subsets MUST be labeled with an appropriate charset value. See
   section 3.4.1 for compatibility problems.

Reply:
Fixed!